TESTS/mbed_hal/ticker_api_tests/main.cpp

```diff
@@ -0,0 +1,520 @@
+ */
+void ticker_increment_test(void)
+{
+    /* Init the ticker. */
+    intf->init();
+
+    /* Get ticker info. */
+    const ticker_info_t* p_ticker_info = intf->get_info();
+
+    /* Perform test based on ticker speed. */
+    if (p_ticker_info->frequency <= 250000) {    // low frequency tickers
+        const uint32_t base_tick_count = intf->read();
+        uint32_t next_tick_count = base_tick_count;
+
+        while (next_tick_count == base_tick_count) {
+            next_tick_count = intf->read();
+        }
+
+        TEST_ASSERT_UINT32_WITHIN(1, next_tick_count, base_tick_count);
+    } else {  // high frequency tickers
+        const uint32_t base_tick_count = count_ticks(NUM_OF_CYCLES, 1);
+        uint32_t next_tick_count = base_tick_count;
+        uint32_t inc_val = 0;
+
+        while (next_tick_count == base_tick_count) {
+            next_tick_count = count_ticks(NUM_OF_CYCLES + inc_val, 1);
+            inc_val++;
+        }
+
+        /* Since we are here we know that next_tick_count != base_tick_count.
+         * The accuracy of our measurement method is +/- 1 tick, so it is possible that
+         * next_tick_count == base_tick_count - 1. This is also valid result.
+         */
+        TEST_ASSERT_UINT32_WITHIN(1, next_tick_count, base_tick_count);
+
+    }
+}
+
+/* Test that common ticker functions complete with the required amount of time.
+ *
+ * Given ticker is available.
+ * When ticker_read, ticker_clear_interrupt, ticker_set_interrupt, ticker_fire_interrupt or ticker_disable_interrupt function is called.
+ * Then its execution is not longer than 20 us.
+ */
+void ticker_speed_test(void)
+{
+    Timer timer;
+    int counter = NUM_OF_CALLS;
+
+    /* Init the ticker. */
+    intf->init();
+
+    /* ---- Test ticker_read function. ---- */
+    timer.reset();
+    timer.start();
+    while (counter--) {
+        intf->read();
+    }
+    timer.stop();
+
+    /* Check result. */
+    TEST_ASSERT(timer.read_us() < (NUM_OF_CALLS * (MAX_FUNC_EXEC_TIME_US + DELTA_FUNC_EXEC_TIME_US)));
+
+    /* ---- Test ticker_clear_interrupt function. ---- */
+    counter = NUM_OF_CALLS;
+    timer.reset();
+    timer.start();
+    while (counter--) {
+        intf->clear_interrupt();
+    }
+    timer.stop();
+
+    /* Check result. */
+    TEST_ASSERT(timer.read_us() < (NUM_OF_CALLS * (MAX_FUNC_EXEC_TIME_US + DELTA_FUNC_EXEC_TIME_US)));
+
+    /* ---- Test ticker_set_interrupt function. ---- */
+    counter = NUM_OF_CALLS;
+    timer.reset();
+    timer.start();
+    while (counter--) {
+        intf->set_interrupt(0);
+    }
+    timer.stop();
+
+    /* Check result. */
+    TEST_ASSERT(timer.read_us() < (NUM_OF_CALLS * (MAX_FUNC_EXEC_TIME_US + DELTA_FUNC_EXEC_TIME_US)));
+
+    /* ---- Test fire_interrupt function. ---- */
+    counter = NUM_OF_CALLS;
+    timer.reset();
+    timer.start();
+    while (counter--) {
+        intf->fire_interrupt();
+    }
+    timer.stop();
+
+    /* Check result. */
```

Comment:
redundant, please review the tests and possibly reduce comments like this

Reply:
Removed redundant comments.